src/protocol/attribution/mod.rs

```diff
@@ -58,100 +58,88 @@ where
 ///
 pub async fn prefix_or_binary_tree_style<F, C, S>(
     ctx: C,
-    helper_bits: &[S],
+    stop_bits: &[S],
     uncapped_credits: &[S],
 ) -> Result<Vec<S>, Error>
 where
     F: Field,
     C: Context,
     S: ArithmeticSecretSharing<F> + BasicProtocols<C, F>,
 {
-    assert_eq!(helper_bits.len() + 1, uncapped_credits.len());
+    assert_eq!(stop_bits.len() + 1, uncapped_credits.len());
 
     let num_rows = uncapped_credits.len();
-    let depth_0_ctx = ctx
-        .narrow(&InteractionPatternStep::from(0))
-        .set_total_records(num_rows - 1);
 
-    let b_times_sibling_credit_ctx = depth_0_ctx.narrow(&Step::BTimesSuccessorCredit);
+    let mut uncapped_credits = uncapped_credits.to_owned();
 
-    let mut prefix_or = try_join_all(
-        helper_bits
-            .iter()
-            .zip(uncapped_credits.iter().skip(1))
-            .enumerate()
-            .map(|(i, (b, sibling_credit))| {
-                let c1 = b_times_sibling_credit_ctx.clone();
-                let c2 = depth_0_ctx.clone();
-                let record_id = RecordId::from(i);
-                let original_credit = &uncapped_credits[i];
-                async move {
-                    let credit_update = S::multiply(c1, record_id, b, sibling_credit).await?;
-                    or(c2, record_id, original_credit, &credit_update).await
-                }
-            }),
-    )
-    .await?;
-    // This is crap and will resize the vector... it was too hard to fix...
-    prefix_or.push(uncapped_credits[num_rows - 1].clone());
-
-    // Create stop_bit vector.
-    // This vector is updated in each iteration to help accumulate values
+    // This vector is updated in each iteration to help accumulate credits
     // and determine when to stop accumulating.
-    let mut stop_bits = helper_bits.to_owned();
-    stop_bits.push(S::share_known_value(&ctx, F::ONE));
+    let mut stop_bits = stop_bits.to_owned();
+    stop_bits.push(S::ZERO);
```

Comment:
I just noticed that this swapped from ONE to ZERO... why is that?

Reply:
This is because of that *semantic change* of the stop bits. We now compute `current_stop_bit * sibling_credit`, and the binary-tree loop skips the bottom N - step_size rows. It doesn't matter what value we store here.

And I just realized we can probably skip the last `current_stop_bit * sibling_stop_bit multiplication` in each depth, because it'll always be 0.